src/server/db/models/github-user.js

```diff
@@ -7,9 +7,32 @@ export default function register(db) {
    *   created_at: creation date
    *   updated_at: update date
    *   last_login_at: last login date
+   *   snaps_added: number of snaps added
+   *   snaps_removed: number of snaps removed
+   *   names_registered: number of snap names registered
+   *   builds_requested: number of snap builds requested
    */
   db.model('GitHubUser', {
     tableName: 'github_user',
     hasTimestamps: true
+  }, {
+    // Increment a metric for a user.  Where possible, this should be called
+    // at the *start* of a transaction spanning any external API calls that
+    // are tracked by the metric in question: this means that either a
+    // failure to increment the metric or a failure in the external API call
+    // will cause both to be rolled back.
+    incrementMetric: async (query, metricName, delta, options) => {
+      const row = await db.model('GitHubUser').where(query).fetch(options);
+      if (row) {
+        const value = row.get(metricName);
+        // Metrics may often require non-trivial initialization, for example
+        // when a metric is introduced to the site after the feature that
+        // it's measuring.  To cope with this, only increment a metric if it
+        // has already been initialized.
+        if (value !== undefined && value !== null) {
```

Comment:
In theory this could be shortened to `value != null` because `undefined == null` (note double not tripple eq), but I'm not sure if we want to have the an exception from using `===` just for this null comparison case.